Examples/minimum_user_for_s2e/src/src_user/Test/test/src_core/System/TimeManager/test_time_manager.py

```diff
@@ -51,57 +53,193 @@ def test_tmgr_set_time():
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+    _test_utl_cmd(unixtime_at_ti0, TMGR_DEFAULT_UNIXTIME_EPOCH_FOR_UTL)
+
+
+    # unixtime_at_ti0 < epoch の場合（意図した TI で UTL が打てない）
+    unixtime_at_ti0 = TMGR_DEFAULT_UNIXTIME_EPOCH_FOR_UTL - 100
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope,
+        c2a_enum.Cmd_CODE_TMGR_SET_UNIXTIME,
+        (unixtime_at_ti0, 0, 0),
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+    _test_utl_cmd(unixtime_at_ti0, TMGR_DEFAULT_UNIXTIME_EPOCH_FOR_UTL)
+
+
+    # epoch が変わった場合
+    new_epoch = time.time()
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope, c2a_enum.Cmd_CODE_TMGR_SET_UTL_UNIXTIME_EPOCH, (new_epoch,), c2a_enum.Tlm_CODE_HK
+    )
+
+    unixtime_at_ti0 = time.time() + 100
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope,
+        c2a_enum.Cmd_CODE_TMGR_SET_UNIXTIME,
+        (unixtime_at_ti0, 0, 0),
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+    _test_utl_cmd(unixtime_at_ti0, new_epoch)
+
+
+@pytest.mark.sils
+@pytest.mark.real
+def test_tmgr_final_check():
+    # unixtime_at_ti0 を初期化する
+    unixtime_at_ti0 = 0.0
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope,
+        c2a_enum.Cmd_CODE_TMGR_SET_UNIXTIME,
+        (unixtime_at_ti0, 0, 0),
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+    # epochをデフォルトに戻す
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope, c2a_enum.Cmd_CODE_TMGR_SET_UTL_UNIXTIME_EPOCH, (TMGR_DEFAULT_UNIXTIME_EPOCH_FOR_UTL,), c2a_enum.Tlm_CODE_HK
+    )
+
+
+def _test_utl_cmd(unixtime_at_ti0, utl_unixtime_epoch):
+    # 最初にTL0をクリアしておく
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope,
+        c2a_enum.Cmd_CODE_TLCD_CLEAR_ALL_TIMELINE,
+        (0, ),
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+    tlm_HK = wings.util.generate_and_receive_tlm(
+        ope, c2a_enum.Cmd_CODE_GENERATE_TLM, c2a_enum.Tlm_CODE_HK
+    )
+    unixtime_now = unixtime_at_ti0 + tlm_HK["HK.SH.TI"] / OBCT_CYCLES_PER_SEC
+
+    # NOP を10個, 未来のランダムな unixtime で登録する
+    unixtime_of_cmds = generate_random_unixtime(unixtime_now)
+    send_utl_nops(unixtime_of_cmds)
+
+    # 重複を削除して時刻順に並べ替え
+    unixtime_of_cmds = list(set(unixtime_of_cmds))
+    unixtime_of_cmds.sort()
+
+
+    # TL0 に正しく登録されているか確認
+    tlm_TL = wings.util.generate_and_receive_tlm(
+        ope, c2a_enum.Cmd_CODE_GENERATE_TLM, c2a_enum.Tlm_CODE_TL
+    )
+    assert tlm_TL["TL.LINE_NO"] == 0
+
+    for i, unixtime in enumerate(unixtime_of_cmds):
+        tlm_name = "TL.CMD" + str(i) + "_TI"
+        ti = calc_ti_from_unixtime(unixtime, unixtime_at_ti0, utl_unixtime_epoch)
+
+        assert tlm_TL[tlm_name] > ti - 1
+        assert tlm_TL[tlm_name] < ti + 1
+
+    # 最後にTL0をもう一度クリアする
+    assert "SUC" == wings.util.send_cmd_and_confirm(
+        ope,
+        c2a_enum.Cmd_CODE_TLCD_CLEAR_ALL_TIMELINE,
+        (0, ),
+        c2a_enum.Tlm_CODE_HK,
+    )
+
+
+# 未来のランダムな時刻の unixtime を 10 個生成する
+def generate_random_unixtime(unixtime_now):
```

Comment:
[IMO] これ，関数名このままなら， `10個` を引数に取るほうがいい？

```suggestion
def generate_random_unixtime(unixtime_now, num):
```
みたいな